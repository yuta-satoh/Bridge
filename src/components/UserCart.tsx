import useSWR, { Fetcher } from "swr";
import Link from "next/link";
import Image from "next/image";
import Recommend from "./Recommend";
import deleteCart from "@/lib/deleteCart";
import { ChangeEvent } from "react";

type Item = {
  id: number,
	name: string,
	description: string,
	genre: string,
	category: string,
	price: number,
	imgurl: string,
	stock: number,
	delete: boolean,
}

type Cart = {
  id: number,
  items: Item,
  cart_id: number,
  date: string,
  quantity: number,
  delete: boolean,
}

const fetcher: Fetcher<Cart[], string> = (...args) => fetch(...args).then((res) => res.json());

export default function UserCart({ userId }: { userId: string }) {
  // SWRでアイテムを取得
  const { data: cartItemData, error, mutate } = useSWR(`/api/getCart/items?id=${userId}`, fetcher)

  // カートにデータがない時の表示
	if (error) return (
    <div className="w-4/5 mx-auto">
      <h2 className="text-3xl font-bold mt-10">カート</h2>
      <div className="block py-40 mx-auto my-10 bg-neutral-100 rounded">
        <h2 className="text-center text-3xl font-bold mb-10">カートに商品がありません</h2>
        <Link href="/">
          <div className="container mx-auto pt-1.5 text-center h-10 w-1/3 border-2 border-neutral-900 bg-white mt-4">
            お買い物はこちらから
          </div>
        </Link>
      </div>
    </div>
  )

	if (!cartItemData) return <div>loading...</div>

  const filtercartItems = cartItemData.filter((cart) => cart.delete === false)

  const recommend = filtercartItems.map((cart) => {
    return {
      id: cart.items.id,
      genre: cart.items.genre
    }
  })
  console.log(recommend);

  // 合計金額
  const sumPrice = filtercartItems.reduce((current, cart) => current + (cart.items.price * cart.quantity), 0)

  const handleDelete = async (itemId: number, cartId: number) => {
    console.log(itemId, cartId);
    await deleteCart(itemId, cartId);
    mutate(filtercartItems);
  }

  const handleChange = async (ev: ChangeEvent<HTMLSelectElement>, itemId: number, cartId: number) => {
    const value = ev.target.value
    console.log(value)
    await fetch("/api/selectQuantity", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        item_id: itemId,
        cart_id: cartId,
        quantity: Number(value),
      })
    }).then((res) => console.log("ステータス:", res.status))
    mutate(filtercartItems)
  }

	return (
    <>
      {filtercartItems.length !== 0 ? (
        <div className="flex mx-auto w-4/5 gap-32">
          <div className="w-3/5 mt-10">
            <div className="flex">
              <h2 className="text-3xl font-bold">カート</h2>
              <span className="mt-2">{`(${filtercartItems.length}点の商品)`}</span>
            </div>
            {filtercartItems.map((cart, index) => (
              <div key={index} className="border border-neutral-900 my-2 py-3 px-8 h-52">
                <div className="flex gap-5">
                  <Link href={`/items/itemlist/${cart.items.id}`}>
                    <Image src={cart.items.imgurl} alt={cart.items.name} width={150} height={150} className="rounded"/>
                  </Link>
                  <div className="px-3 py-4">
                    <Link href={`/items/itemlist/${cart.items.id}`}>
                      <p className="underline mb-1 text-xl">{cart.items.name}</p>
                    </Link>
                    <p className="text-sm mt-1">{cart.items.description}</p>
                    <p className="mt-1 text-lg">¥ {cart.items.price.toLocaleString()}</p>
                  </div>
                </div>
                <div className="text-right">
                  <button
                    className="text-white bg-neutral-900 border border-neutral-900 rounded px-1"
                    onClick={() => handleDelete(cart.items.id, cart.cart_id)}  
                  >
                    削除
                  </button>
                  {/* 個数を変えたら金額も変更したい（CSRで） */}
                  <select
                    name="quantity"
                    id="cart_quantity"
                    className="ml-5 border border-neutral-900 rounded p-1"
                    defaultValue={cart.quantity}
                    onChange={(ev) => handleChange(ev, cart.items.id, cart.cart_id)}
                  >
                    <option value={1}>1</option>
                    <option value={2}>2</option>
                    <option value={3}>3</option>
                    <option value={4}>4</option>
                    <option value={5}>5</option>
                    <option value={6}>6</option>
                    <option value={7}>7</option>
                    <option value={8}>8</option>
                    <option value={9}>9</option>
                    <option value={10}>10</option>
                  </select>
                </div>
              </div>  
            ))}
            <Recommend recommend={recommend} />
          </div>
          <div className="w-1/4 h-80 mt-20 p-10 border-2 border-neutral-900 rounded bg-gray-100 sticky top-10">
            <p>
              <span>商品合計</span>
              <span className="float-right">¥ {sumPrice.toLocaleString()}</span>
            </p>
            <p>
              <span>消費税</span>
              <span className="float-right">¥ {(sumPrice * 0.1).toLocaleString()}</span>
            </p>
            <div className="border-b border-black">
              <p className="mt-10 text-lg font-bold">
                <span>合計</span>
                <span className="float-right">¥ {(sumPrice + (sumPrice * 0.1)).toLocaleString()}</span>
              </p>
            </div>
            <Link href="/purchase">
              <div className="container pt-1.5 text-center h-10 border-2 border-neutral-900 bg-white mt-8">
                <span>ご注文手続き</span>
              </div>
            </Link>
            <Link href="/">
              <div className="container pt-1.5 text-center h-10 border-2 border-neutral-900 bg-white mt-4">
                <span>お買い物を続ける</span>
              </div>
            </Link>
          </div>
        </div>
      ) : (
        <div className="w-4/5 mx-auto">
          <h2 className="text-3xl font-bold mt-10">カート</h2>
          <div className="block py-40 mx-auto my-10 bg-neutral-100 rounded">
            <h2 className="text-center text-3xl font-bold mb-10">カートに商品がありません</h2>
            <Link href="/">
              <div className="container mx-auto pt-1.5 text-center h-10 w-1/3 border-2 border-neutral-900 bg-white mt-4">
                お買い物はこちらから
              </div>
            </Link>
          </div>
        </div>
      )}
    </>
  )
}
