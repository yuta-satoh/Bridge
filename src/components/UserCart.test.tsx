import { render, screen } from "@testing-library/react";
import { Middleware, SWRResponse, SWRConfig } from "swr";
import userEvent from "@testing-library/user-event";

import UserCart from "./UserCart";

jest.mock('next/router', () => ({
    useRouter() {
      return {
        replace: jest.fn(),
        push: jest.fn(),
        reload: jest.fn(),
      };
    },
}));

jest.mock('./Recommend', () => {{
    return 'div'
}})

let assignMock = jest.fn();

const fetchMockOk = () => {
    return new Promise((resolve) => {
        resolve({
            ok: true,
            status: 200,
            statusText: 'OK',
            json: () => '1'
        })
    })
}

describe('ユーザーカートコンポーネント', () => {
    test('ユーザーカートが空の状態', () => {
        const cartMiddleware: Middleware = () => {
            return (): SWRResponse<any, any> => {
                return {
                    data: [],
                    error: Error(),
                    mutate: (_) => Promise.resolve(),
                    isValidating: false,
                    isLoading: false,
                }
            }
        }
        const { asFragment } = render(
            <SWRConfig value={{ use: [cartMiddleware] }}>
                <UserCart userId="1" />
            </SWRConfig>
        );
        expect(asFragment()).toMatchSnapshot();
    })

    test('ユーザーカートをローディング中', () => {
        const cartMiddleware: Middleware = () => {
            return (): SWRResponse<any, any> => {
                return {
                    data: undefined,
                    error: undefined,
                    mutate: (_) => Promise.resolve(),
                    isValidating: false,
                    isLoading: false,
                }
            }
        }
        const { asFragment } = render(
            <SWRConfig value={{ use: [cartMiddleware] }}>
                <UserCart userId="1" />
            </SWRConfig>
        );
        expect(asFragment()).toMatchSnapshot();
    })

    test('ユーザーカートに中身あり', () => {
        const cartMiddleware: Middleware = () => {
            return (): SWRResponse<any, any> => {
                const mockData = [
                    {
                        id: 1,
                        items: {
                            id: 1,
                            name: "ベルベットアームチェア",
                            description: "ベルベットのお洒落なデザインチェア。",
                            genre: "フェミニン",
                            category: "椅子",
                            price: 16000,
                            imgurl: "/images/chair/chair_feminine_1.jpg",
                            stock: 72,
                            delete: false
                        },
                        cart_id: 1,
                        date: '2023-03-13',
                        quantity: 1,
                        delete: false,
                    },
                    {
                        id: 2,
                        items: {
                            id: 2,
                            name: "丸椅子スツール",
                            description: "落ち着いた可愛らしいベロア風スツール。",
                            genre: "フェミニン",
                            category: "椅子",
                            price: 10000,
                            imgurl: "/images/chair/chair_feminine_2.jpg",
                            stock: 99,
                            delete: false
                        }, 
                        cart_id: 1,
                        date: '2023-03-13',
                        quantity: 1,
                        delete: false,
                    },
                ];
                
                return {
                    data: mockData,
                    error: undefined,
                    mutate: (_) => Promise.resolve(),
                    isValidating: false,
                    isLoading: false,
                }
            }
        }
        const { asFragment } = render(
            <SWRConfig value={{ use: [cartMiddleware] }}>
                <UserCart userId="1" />
            </SWRConfig>
        );
        expect(asFragment()).toMatchSnapshot();
    })

    test('ユーザーカートの商品の個数を変更する', async () => {
        global.fetch = jest.fn().mockImplementation(fetchMockOk);
        const user = userEvent.setup()

        // 初期レンダリング用モックデータ
        const cartMiddleware: Middleware = () => {
            return (): SWRResponse<any, any> => {
                const mockData = [
                    {
                        id: 1,
                        items: {
                            id: 1,
                            name: "ベルベットアームチェア",
                            description: "ベルベットのお洒落なデザインチェア。",
                            genre: "フェミニン",
                            category: "椅子",
                            price: 16000,
                            imgurl: "/images/chair/chair_feminine_1.jpg",
                            stock: 72,
                            delete: false
                        },
                        cart_id: 1,
                        date: '2023-03-13',
                        quantity: 1,
                        delete: false,
                    },
                ];
                
                return {
                    data: mockData,
                    error: undefined,
                    mutate: (_) => Promise.resolve(),
                    isValidating: false,
                    isLoading: false,
                }
            }
        }

        // 個数変更後のモックデータ
        const changeMiddleware: Middleware = () => {
            return (): SWRResponse<any, any> => {
                const mockData = [
                    {
                        id: 1,
                        items: {
                            id: 1,
                            name: "ベルベットアームチェア",
                            description: "ベルベットのお洒落なデザインチェア。",
                            genre: "フェミニン",
                            category: "椅子",
                            price: 16000,
                            imgurl: "/images/chair/chair_feminine_1.jpg",
                            stock: 72,
                            delete: false
                        },
                        cart_id: 1,
                        date: '2023-03-13',
                        quantity: 2,
                        delete: false,
                    },
                ];
                
                return {
                    data: mockData,
                    error: undefined,
                    mutate: (_) => Promise.resolve(),
                    isValidating: false,
                    isLoading: false,
                }
            }
        }
        const {asFragment, rerender} = render(
            <SWRConfig value={{ use: [cartMiddleware] }}>
                <UserCart userId="1" />
            </SWRConfig>
        );
        const selectId = screen.getByTestId("select-option");
        await user.selectOptions(selectId, '2');
        // セレクトボックスは変更できたが金額は反映されず
        // 再レンダリングして個数を変更したデータを渡す
        rerender(
            <SWRConfig value={{ use: [changeMiddleware] }}>
                <UserCart userId="1" />
            </SWRConfig>
        )
        expect(asFragment()).toMatchSnapshot();
    })

    test('ユーザーカートの商品を削除', async () => {
        global.fetch = jest.fn().mockImplementation(fetchMockOk);
        const user = userEvent.setup();
        const cartMiddleware: Middleware = () => {
            return (): SWRResponse<any, any> => {
                const mockData = [
                    {
                        id: 1,
                        items: {
                            id: 1,
                            name: "ベルベットアームチェア",
                            description: "ベルベットのお洒落なデザインチェア。",
                            genre: "フェミニン",
                            category: "椅子",
                            price: 16000,
                            imgurl: "/images/chair/chair_feminine_1.jpg",
                            stock: 72,
                            delete: false
                        },
                        cart_id: 1,
                        date: '2023-03-13',
                        quantity: 1,
                        delete: false,
                    },
                    {
                        id: 2,
                        items: {
                            id: 2,
                            name: "丸椅子スツール",
                            description: "落ち着いた可愛らしいベロア風スツール。",
                            genre: "フェミニン",
                            category: "椅子",
                            price: 10000,
                            imgurl: "/images/chair/chair_feminine_2.jpg",
                            stock: 99,
                            delete: false
                        }, 
                        cart_id: 1,
                        date: '2023-03-13',
                        quantity: 1,
                        delete: false,
                    },
                ];
                
                return {
                    data: mockData,
                    error: undefined,
                    mutate: (_) => Promise.resolve(),
                    isValidating: false,
                    isLoading: false,
                }
            }
        }
        const deleteMiddleware: Middleware = () => {
            return (): SWRResponse<any, any> => {
                const mockData = [
                    {
                        id: 1,
                        items: {
                            id: 1,
                            name: "ベルベットアームチェア",
                            description: "ベルベットのお洒落なデザインチェア。",
                            genre: "フェミニン",
                            category: "椅子",
                            price: 16000,
                            imgurl: "/images/chair/chair_feminine_1.jpg",
                            stock: 72,
                            delete: false
                        },
                        cart_id: 1,
                        date: '2023-03-13',
                        quantity: 1,
                        delete: false,
                    },
                ];
                
                return {
                    data: mockData,
                    error: undefined,
                    mutate: (_) => Promise.resolve(),
                    isValidating: false,
                    isLoading: false,
                }
            }
        }
        const { asFragment, rerender } = render(
            <SWRConfig value={{ use: [cartMiddleware] }}>
                <UserCart userId="1" />
            </SWRConfig>
        );
        const deleteBtn = screen.getAllByRole('button');
        await user.click(deleteBtn[1]);
        // 再レンダリングして削除後のデータを渡す
        rerender(
            <SWRConfig value={{ use: [deleteMiddleware] }}>
                <UserCart userId="1" />
            </SWRConfig>    
        )
        expect(asFragment()).toMatchSnapshot();
    })
})
