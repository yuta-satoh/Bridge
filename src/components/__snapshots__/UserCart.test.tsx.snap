// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ユーザーカートコンポーネント ユーザーカートが空の状態 1`] = `
<DocumentFragment>
  <div
    class="w-4/5 mx-auto"
  >
    <h2
      class="text-3xl font-bold mt-10"
    >
      カート
    </h2>
    <div
      class="block py-40 mx-auto my-10 bg-neutral-100 rounded"
    >
      <h2
        class="text-center text-3xl font-bold mb-10"
      >
        カートに商品がありません
      </h2>
      <a
        href="/"
      >
        <div
          class="container mx-auto pt-1.5 text-center h-10 w-1/3 border-2 border-neutral-900 bg-white mt-4"
        >
          お買い物はこちらから
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ユーザーカートコンポーネント ユーザーカートに中身あり 1`] = `
<DocumentFragment>
  <div
    class="flex mx-auto w-4/5 gap-32"
  >
    <div
      class="w-3/5 mt-10"
    >
      <div
        class="flex"
      >
        <h2
          class="text-3xl font-bold"
        >
          カート
        </h2>
        <span
          class="mt-2"
        >
          (2点の商品)
        </span>
      </div>
      <div
        class="border border-neutral-900 my-2 py-3 px-8 h-52"
      >
        <div
          class="flex gap-5"
        >
          <a
            href="/items/itemlist/1"
          >
            <img
              alt="ベルベットアームチェア"
              class="rounded"
              data-nimg="1"
              decoding="async"
              height="150"
              loading="lazy"
              src="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=384&q=75"
              srcset="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=256&q=75 1x, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=384&q=75 2x"
              style="color: transparent;"
              width="150"
            />
          </a>
          <div
            class="px-3 py-4"
          >
            <a
              href="/items/itemlist/1"
            >
              <p
                class="underline mb-1 text-xl"
              >
                ベルベットアームチェア
              </p>
            </a>
            <p
              class="text-sm mt-1"
            >
              ベルベットのお洒落なデザインチェア。
            </p>
            <p
              class="mt-1 text-lg"
            >
              ¥ 16,000
            </p>
          </div>
        </div>
        <div
          class="text-right"
        >
          <button
            class="text-white bg-neutral-900 border border-neutral-900 rounded px-1"
          >
            削除
          </button>
          <select
            class="ml-5 border border-neutral-900 rounded p-1"
            data-testid="select-option"
            id="cart_quantity"
            name="quantity"
          >
            <option
              selected=""
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6"
            >
              6
            </option>
            <option
              value="7"
            >
              7
            </option>
            <option
              value="8"
            >
              8
            </option>
            <option
              value="9"
            >
              9
            </option>
            <option
              value="10"
            >
              10
            </option>
          </select>
        </div>
      </div>
      <div
        class="border border-neutral-900 my-2 py-3 px-8 h-52"
      >
        <div
          class="flex gap-5"
        >
          <a
            href="/items/itemlist/2"
          >
            <img
              alt="丸椅子スツール"
              class="rounded"
              data-nimg="1"
              decoding="async"
              height="150"
              loading="lazy"
              src="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_2.jpg&w=384&q=75"
              srcset="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_2.jpg&w=256&q=75 1x, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_2.jpg&w=384&q=75 2x"
              style="color: transparent;"
              width="150"
            />
          </a>
          <div
            class="px-3 py-4"
          >
            <a
              href="/items/itemlist/2"
            >
              <p
                class="underline mb-1 text-xl"
              >
                丸椅子スツール
              </p>
            </a>
            <p
              class="text-sm mt-1"
            >
              落ち着いた可愛らしいベロア風スツール。
            </p>
            <p
              class="mt-1 text-lg"
            >
              ¥ 10,000
            </p>
          </div>
        </div>
        <div
          class="text-right"
        >
          <button
            class="text-white bg-neutral-900 border border-neutral-900 rounded px-1"
          >
            削除
          </button>
          <select
            class="ml-5 border border-neutral-900 rounded p-1"
            data-testid="select-option"
            id="cart_quantity"
            name="quantity"
          >
            <option
              selected=""
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6"
            >
              6
            </option>
            <option
              value="7"
            >
              7
            </option>
            <option
              value="8"
            >
              8
            </option>
            <option
              value="9"
            >
              9
            </option>
            <option
              value="10"
            >
              10
            </option>
          </select>
        </div>
      </div>
      <div
        recommend="[object Object],[object Object]"
      />
    </div>
    <div
      class="w-1/4 h-80 mt-20 p-10 border-2 border-neutral-900 rounded bg-gray-100 sticky top-10"
    >
      <p>
        <span>
          商品合計
        </span>
        <span
          class="float-right"
        >
          ¥ 26,000
        </span>
      </p>
      <p>
        <span>
          消費税
        </span>
        <span
          class="float-right"
        >
          ¥ 2,600
        </span>
      </p>
      <div
        class="border-b border-black"
      >
        <p
          class="mt-10 text-lg font-bold"
        >
          <span>
            合計
          </span>
          <span
            class="float-right"
          >
            ¥ 28,600
          </span>
        </p>
      </div>
      <a
        href="/purchase"
      >
        <div
          class="container pt-1.5 text-center h-10 border-2 border-neutral-900 bg-white mt-8"
        >
          <span>
            ご注文手続き
          </span>
        </div>
      </a>
      <a
        href="/"
      >
        <div
          class="container pt-1.5 text-center h-10 border-2 border-neutral-900 bg-white mt-4"
        >
          <span>
            お買い物を続ける
          </span>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ユーザーカートコンポーネント ユーザーカートの商品の個数を変更する 1`] = `
<DocumentFragment>
  <div
    class="flex mx-auto w-4/5 gap-32"
  >
    <div
      class="w-3/5 mt-10"
    >
      <div
        class="flex"
      >
        <h2
          class="text-3xl font-bold"
        >
          カート
        </h2>
        <span
          class="mt-2"
        >
          (1点の商品)
        </span>
      </div>
      <div
        class="border border-neutral-900 my-2 py-3 px-8 h-52"
      >
        <div
          class="flex gap-5"
        >
          <a
            href="/items/itemlist/1"
          >
            <img
              alt="ベルベットアームチェア"
              class="rounded"
              data-nimg="1"
              decoding="async"
              height="150"
              loading="lazy"
              src="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=384&q=75"
              srcset="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=256&q=75 1x, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=384&q=75 2x"
              style="color: transparent;"
              width="150"
            />
          </a>
          <div
            class="px-3 py-4"
          >
            <a
              href="/items/itemlist/1"
            >
              <p
                class="underline mb-1 text-xl"
              >
                ベルベットアームチェア
              </p>
            </a>
            <p
              class="text-sm mt-1"
            >
              ベルベットのお洒落なデザインチェア。
            </p>
            <p
              class="mt-1 text-lg"
            >
              ¥ 16,000
            </p>
          </div>
        </div>
        <div
          class="text-right"
        >
          <button
            class="text-white bg-neutral-900 border border-neutral-900 rounded px-1"
          >
            削除
          </button>
          <select
            class="ml-5 border border-neutral-900 rounded p-1"
            data-testid="select-option"
            id="cart_quantity"
            name="quantity"
          >
            <option
              selected=""
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6"
            >
              6
            </option>
            <option
              value="7"
            >
              7
            </option>
            <option
              value="8"
            >
              8
            </option>
            <option
              value="9"
            >
              9
            </option>
            <option
              value="10"
            >
              10
            </option>
          </select>
        </div>
      </div>
      <div
        recommend="[object Object]"
      />
    </div>
    <div
      class="w-1/4 h-80 mt-20 p-10 border-2 border-neutral-900 rounded bg-gray-100 sticky top-10"
    >
      <p>
        <span>
          商品合計
        </span>
        <span
          class="float-right"
        >
          ¥ 32,000
        </span>
      </p>
      <p>
        <span>
          消費税
        </span>
        <span
          class="float-right"
        >
          ¥ 3,200
        </span>
      </p>
      <div
        class="border-b border-black"
      >
        <p
          class="mt-10 text-lg font-bold"
        >
          <span>
            合計
          </span>
          <span
            class="float-right"
          >
            ¥ 35,200
          </span>
        </p>
      </div>
      <a
        href="/purchase"
      >
        <div
          class="container pt-1.5 text-center h-10 border-2 border-neutral-900 bg-white mt-8"
        >
          <span>
            ご注文手続き
          </span>
        </div>
      </a>
      <a
        href="/"
      >
        <div
          class="container pt-1.5 text-center h-10 border-2 border-neutral-900 bg-white mt-4"
        >
          <span>
            お買い物を続ける
          </span>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ユーザーカートコンポーネント ユーザーカートの商品を削除 1`] = `
<DocumentFragment>
  <div
    class="flex mx-auto w-4/5 gap-32"
  >
    <div
      class="w-3/5 mt-10"
    >
      <div
        class="flex"
      >
        <h2
          class="text-3xl font-bold"
        >
          カート
        </h2>
        <span
          class="mt-2"
        >
          (1点の商品)
        </span>
      </div>
      <div
        class="border border-neutral-900 my-2 py-3 px-8 h-52"
      >
        <div
          class="flex gap-5"
        >
          <a
            href="/items/itemlist/1"
          >
            <img
              alt="ベルベットアームチェア"
              class="rounded"
              data-nimg="1"
              decoding="async"
              height="150"
              loading="lazy"
              src="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=384&q=75"
              srcset="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=256&q=75 1x, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=384&q=75 2x"
              style="color: transparent;"
              width="150"
            />
          </a>
          <div
            class="px-3 py-4"
          >
            <a
              href="/items/itemlist/1"
            >
              <p
                class="underline mb-1 text-xl"
              >
                ベルベットアームチェア
              </p>
            </a>
            <p
              class="text-sm mt-1"
            >
              ベルベットのお洒落なデザインチェア。
            </p>
            <p
              class="mt-1 text-lg"
            >
              ¥ 16,000
            </p>
          </div>
        </div>
        <div
          class="text-right"
        >
          <button
            class="text-white bg-neutral-900 border border-neutral-900 rounded px-1"
          >
            削除
          </button>
          <select
            class="ml-5 border border-neutral-900 rounded p-1"
            data-testid="select-option"
            id="cart_quantity"
            name="quantity"
          >
            <option
              selected=""
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6"
            >
              6
            </option>
            <option
              value="7"
            >
              7
            </option>
            <option
              value="8"
            >
              8
            </option>
            <option
              value="9"
            >
              9
            </option>
            <option
              value="10"
            >
              10
            </option>
          </select>
        </div>
      </div>
      <div
        recommend="[object Object]"
      />
    </div>
    <div
      class="w-1/4 h-80 mt-20 p-10 border-2 border-neutral-900 rounded bg-gray-100 sticky top-10"
    >
      <p>
        <span>
          商品合計
        </span>
        <span
          class="float-right"
        >
          ¥ 16,000
        </span>
      </p>
      <p>
        <span>
          消費税
        </span>
        <span
          class="float-right"
        >
          ¥ 1,600
        </span>
      </p>
      <div
        class="border-b border-black"
      >
        <p
          class="mt-10 text-lg font-bold"
        >
          <span>
            合計
          </span>
          <span
            class="float-right"
          >
            ¥ 17,600
          </span>
        </p>
      </div>
      <a
        href="/purchase"
      >
        <div
          class="container pt-1.5 text-center h-10 border-2 border-neutral-900 bg-white mt-8"
        >
          <span>
            ご注文手続き
          </span>
        </div>
      </a>
      <a
        href="/"
      >
        <div
          class="container pt-1.5 text-center h-10 border-2 border-neutral-900 bg-white mt-4"
        >
          <span>
            お買い物を続ける
          </span>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ユーザーカートコンポーネント ユーザーカートをローディング中 1`] = `
<DocumentFragment>
  <div>
    loading...
  </div>
</DocumentFragment>
`;
